### DISCUSSION TITLE
400 Bad Request

### DISCUSSION THREAD

**Discussion by lukibazuki** [astrandb/miele#153] [Q&A] [answered]
I got everything working and can monitor the washing machine while its running, however, i can not control it through HA. I always get the 400 Bad Request failure. 

You’ll see on the picture what I have been trying. It does not work through automation or developer tools. 

![6D9CAE86-1B36-4B29-9DDC-49A8C07206DE](https://user-images.githubusercontent.com/24279405/200111785-7d3b0a65-3778-495b-a2f5-1cc0d83358f3.jpeg)

Help is greatly appreciated.

**Comment by Waschbeton** [comment]
I have the exact same problem, even getting 400 error on the swagger.

{
  "code": 400,
  "message": "Response from setting program was not OK or OK but RESPONSE_FROM_DEVICE_ERROR instead."
}

**Comment by astrandb** [comment, marked as answer]
The 400 error indicates that either the API or the machine itself does not accept the command at this specific moment.

On my washing machine WCI870 the following sequence works:
1. Set the dial to Mobile start
2. Press Start on the touch panel
3. Run `miele.set_program` service from the integration/developer tools

You can see what programs that are available after item 2 above if you do a diagnostic download. Note that the program list will be empty unless the machine is ready to accept a command.

**Comment by friendlycrow** [comment]
I just had this error too. It appears that the problem is that the program id is thrown onto the machine before it is online.

This works:
`alias: daily dishes
description: ""
triggers: []
conditions: []
actions:
  - sequence:
      - type: turn_on
        device_id: 5d4eda265b1e96 (truncated...)
        entity_id: a7af66788dadb71
        domain: switch
        enabled: true
      - wait_template: " {{ (states(\"sensor.dishwasher_status\") == 'programmed') }}"
        continue_on_timeout: true
      - action: miele.set_program
        target:
          device_id: 5d4eda265b1e96
        data:
          programId: 9
mode: single
`
